src/UnityExtension/Assets/Editor/GitHub.Unity/UI/FileHistoryWindow.cs

```diff
@@ -0,0 +1,250 @@
+                fileHistoryWindow.SetSelectedPath(assetPath);
+                fileHistoryWindow.Show();
+            }
+        }
+
+        [MenuItem("Assets/Git/History", true)]
+        private static bool GitFileHistoryValidation()
+        {
+            return Selection.assetGUIDs != null && Selection.assetGUIDs.Length > 0;
+        }
+
+        private const string Title = "File History";
+
+        [NonSerialized] private bool firstOnGUI = true;
+        [NonSerialized] private Texture selectedIcon;
+
+        [SerializeField] private bool locked;
+        [SerializeField] private FileHistoryView fileHistoryView = new FileHistoryView();
+        [SerializeField] private UnityEngine.Object selectedObject;
+        [SerializeField] private string selectedObjectAssetPath;
+        [SerializeField] private string selectedObjectRepositoryPath;
+
+        public void SetSelectedPath(string assetPath)
+        {
+            selectedObject = null;
+            selectedObjectAssetPath = null;
+            selectedObjectRepositoryPath = null;
+
+            if (selectedObjectAssetPath != NPath.Default)
+            {
+                selectedObjectAssetPath = assetPath;
+                selectedObject = AssetDatabase.LoadMainAssetAtPath(selectedObjectAssetPath);
+
+                selectedObjectRepositoryPath =
+                    Environment.GetRepositoryPath(assetPath.ToNPath())
+                               .ToString(SlashMode.Forward);
+            }
+
+            LoadSelectedIcon();
+
+            Repository.UpdateFileLog(selectedObjectRepositoryPath)
+                      .Start();
+        }
+
+        public override void Initialize(IApplicationManager applicationManager)
+        {
+            base.Initialize(applicationManager);
+
+            fileHistoryView.InitializeView(this);
+        }
+
+        public override bool IsBusy
+        {
+            get { return false; }
+        }
+
+        public override void OnEnable()
+        {
+            base.OnEnable();
+
+            LoadSelectedIcon();
+
+            if (fileHistoryView != null)
+                fileHistoryView.OnEnable();
+        }
+
+        public override void OnDisable()
+        {
+            base.OnDisable();
+            if (fileHistoryView != null)
+                fileHistoryView.OnDisable();
+        }
+
+        public override void OnDataUpdate()
+        {
+            base.OnDataUpdate();
+            MaybeUpdateData();
+
+            if (fileHistoryView != null)
+                fileHistoryView.OnDataUpdate();
+        }
+
+        public override void OnRepositoryChanged(IRepository oldRepository)
+        {
+            base.OnRepositoryChanged(oldRepository);
+
+            DetachHandlers(oldRepository);
+            AttachHandlers(Repository);
+
+            if (HasRepository)
+            {
+                
+            }
+            else
+            {
+               
+            }
```

Comment:
This if block had something at some point, I'm guessing?

Reply:
👍

Reply:
Thanks @shana